### DISCUSSION TITLE
Cobrança recorrente de anuidade via Paypal

### DISCUSSION THREAD

**Discussion by rennerocha** [apyb/comunidade#100] [Geral]
**Problema:** Todo ano, a poucas horas da Assembleia na Python Brasil, várias pessoas lembram de efetuar o pagamento da anuidade para se tornar associado, já que nenhum email é enviado para a pessoa quando a anuidade vence (pelo menos eu não lembro de ter recebido nenhum). Como não fui na Python Brasil 2022, eu fiquei um ano inteiro sem pagar a anuidade, simplesmente por ter esquecido!

**Proposta:** Criar um plano de assinaturas recorrente no Paypal (ou serviço semelhante) para que a cobrança seja automática. Desse modo, a possibilidade de alguém esquecer é quase nula, já que vai ser debitada do cartão de crédito automaticamente.

**Alguns pontos que provavelmente serão levantados**:
- A criação de uma conta vinculado ao CNPJ é extremamente simples, a burocracia é pequena, necessitando apenas de documentos básicos
- Isso já foi implementado com sucesso no [Laboratório Hacker de Campinas](https://lhc.net.br), onde as mensalidades são feitas em sua maioria através do Paypal. A indimplência caiu absurdamente. Nós somos uma Associação como a APyB, então sei que não existe nenhum problema contábil com o uso dessa conta.
- As taxas que são cobradas podem parecer altas, porém a redução da inadimplência, diminuição drástica do esquecimento involuntário e a comodidade operacional para os responsávels em gerenciar as assinaturas
- Eu sugiro o Paypal por: permitir assinaturas com cartões de crédito internacionais (facilitando associados que moram fora do Brasil), ter webhooks (permitindo integrações futuras com outros sistemas da APyB)
- O Paypal já teve problemas com "travar" contas e bloquear o dinheiro. Porém é possível facilmente configurar para que qualquer dinheiro na conta seja transferido automaticamente e semanalmente para a conta bancária. Isso minimiza o risco. Caso o Paypal bloqueie a conta, ficaríamos no máximo com uma semana de dinheiro bloqueado.
- Extratos do Paypal são facilmente gerados para transparência
- É possível configurar níveis de acesso para que várias pessoas (autorizadas) tenham permissões no acesso a conta

Caso seja considerado, eu posso acompanhar diretamente a implementação, junto com o responsável da APyB.

**Comment by rougeth** [comment]
Ótima sugestão @rennerocha.

> eu fiquei um ano inteiro sem pagar a anuidade, simplesmente por ter esquecido!

Isso é bem ruim mesmo, tenho certeza que temos vários outros casos assim.

> já que nenhum email é enviado para a pessoa quando a anuidade vence (pelo menos eu não lembro de ter recebido nenhum)

Eu não lembro direito quem refatorou o comando de envio de e-mails com aviso sobre vencimento (talvez @cuducos ou @mazulo), mas desde 2021 eu recebo os avisos. Em 2022 e 2023 foram 3 e-mails, com avisos de 60, 30 e 15 dias para vencimento. O título do e-mail é `[Associação Python Brasil] Aviso de renovação`. Eu vou dar uma olhada nos logs para ver se encontro algo relacionado ao seu caso especificamente.

> Criar um plano de assinaturas recorrente no Paypal

Esse deveria ser o caminho padrão das assinaturas. Hoje usamos Pagseguro, será que eles não oferecem algo semelhante? A opção de receber pagamentos do exterior seria ótima, inclusive, a issue mais antiga do repositório do site é sobre isso https://github.com/apyb/associados/issues/96.

Talvez seja uma oportunidade de revermos os provedores de pagamento, taxas, etc.

**Reply by luzfcb** [reply]
Só para registro, eu venho recebendo os emails de aviso de renovação normalmente desde 2017
![image](https://github.com/apyb/comunidade/assets/807599/64976396-c8b5-4179-a8fb-2809f994c863)

**Reply by rennerocha** [reply]
Eu verifiquei o meu email e realmente estou recebendo notificações. Eu simplesmente ignorei elas, provavelmente por virem no meio de um monte de emails. De qualquer modo, se houver um modo automático, com certeza será o que eu escolherei.

**Comment by rennerocha** [comment]
> Esse deveria ser o caminho padrão das assinaturas. Hoje usamos Pagseguro, será que eles não oferecem algo semelhante? A opção de receber pagamentos do exterior seria ótima, inclusive, a issue mais antiga do repositório do site é sobre isso https://github.com/apyb/associados/issues/96.

Eu acho que o Pagseguro não permite cartões internacionais. Eu pessoalmente também não gosto dele. Nós usavamos Paypal, Pagseguro e MercadoPago no LHC, agora Paypal e raramente MercadoPago.

Eu vi os vários poréns na issue que você citou sobre o Paypal, que eles bloqueiam dinheiro e tal. Eu acredito que isso não seria um problema para gente, já que configurado para sacar todo o dinheiro automaticamente semanalmente, se por um acaso fossemos bloqueados, travaria no máximo uma semana.

Acho que a @cecivieira é a pessoa para levar a discussão e tentar implementar.

**Reply by rougeth** [reply]
> Acho que a @cecivieira é a pessoa para levar a discussão e tentar implementar.

Tivemos várias propostas e ideias como essa ao longo dos últimos dois anos (como tirar o site do Heroku), e nós, da diretoria, não conseguimos implementá-las por falta de tempo.

O melhor cenário seria que as pessoas associadas se envolvessem e tirassem do papel, com o auxílio da diretoria da APyB.

**Reply by rennerocha** [reply]
Eu só marquei ela, pq ela que vai ser a nova responsável financeira da APyB, então vai ter que passar na mão dela de qualquer jeito. Caso o Paypal seja a solução definida, eu posso fazer todo o corre (tendo acesso às informações necessárias da APyB)

**Comment by cuducos** [comment]
> Eu acho que o Pagseguro não permite cartões internacionais.

Não aceita mesmo. Por isso estou inadimplente, por sinal.

**Comment by luzfcb** [comment]
Eu olharia a [Ebanx](https://www.ebanx.com/) como alternativa de meio de pagamento. Ela é uma empresa brasileira. O Spotify BR usa ela para cobrar mensalidade, o Aliexpress usava até uns 2 anos atras, e eu já usei para suportar pagamento em Reais e parcelamento em um site de uma pequena editora americana de livros tecnicos (Feldroy).

**Comment by rennerocha** [comment]
@cecivieira
Houve alguma discussão na diretoria sobre essa questão?
Como falei anteriormente, posso correr para operacionalizar isso, apenas iria precisar de informações burocráticas da APyB.

**Reply by naanadr** [reply]
@rennerocha está no meu backlog, você consegue acompanhar quando eu tiver atualizações sobre isso na issue https://github.com/apyb/tarefas/issues/63 . 

Devo começar a ver isso semana que vem 😄 🙏🏼

**Reply by naanadr** [reply]
@rennerocha não sei se você chegou a ver o nosso último comunicado (https://github.com/apyb/comunidade/discussions/139) , mas mudamos o sistema de pagamento e agora é possível cadastrar cobrança recorrente 😄

**Reply by rennerocha** [reply]
Eu vi :-) Agora não esqueço mais!
